src/cmd.js

```diff
@@ -49,41 +50,67 @@ function handleRunCmd(cmd, dispatch, getState){
   }
 }
 
-export const cmdToPromise = (cmd, dispatch, getState) => {
+function handleParallelList({cmds, batch}, dispatch, getState){
+  const promises = cmds.map(nestedCmd => {
+    const possiblePromise = executeCmd(nestedCmd, dispatch, getState);
+    if(!possiblePromise || batch){
+      return possiblePromise;
+    }
+
+    return possiblePromise.then(result => {
+      return Promise.all(result.map(dispatch));
+    });
+  }).filter(x => x);
+
+  if (promises.length === 0){
+    return null;
+  }
+
+  return Promise.all(promises).then(flatten).then(actions => {
```

Comment:
i made an implementation change here. I realized that the promise returned by dispatch should probably still wait for everything to be finished before resolving, even if it doesn't wait to dispatch any actions. Now it resolves at the end with an empty array of actions to dispatch